lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -1077,6 +1052,34 @@ selectAssets pp utxoAssumptions outs' redeemers
         -- in the final stage of 'balanceTransaction'.
         extraBytes = 8
 
+    transformSelection
+        :: CoinSelection.Selection
+        -> (Selection, changeState)
+    transformSelection sel =
+        let
+            (sel', s') = assignChangeAddresses changeGen sel s
+            inputs = F.toList (sel' ^. #inputs)
+            collateral = sel' ^. #collateral
+            change = sel' ^. #change
+            inputScripts =
+                case utxoAssumptions of
+                    AllKeyPaymentCredentials -> []
+                    AllByronKeyPaymentCredentials -> []
+                    AllScriptPaymentCredentialsFrom _template toInpScripts ->
+                        ( toInpScripts
+                        . Convert.toLedgerAddress
+                        . view #address
+                        . snd
+                        ) <$> inputs <> collateral
+        in  ( Selection
+                { extraInputs = map fst inputs
```

Comment:
👍 I suspect it is true that `extraInputs` have mistakenly contained "all inputs" since the inception of balanceTx. I think I ran into this at some point when debugging a property failure. But it currently causes no damage as adding existing items to the Tx's final set of inputs makes no difference.

I'd prefer leaving this unchanged (but perhaps with a comment) in this PR and fix separately, as a fix could theoretically be risky.

Reply:
@Anviking Fair enough!
Could you add a TODO to the place where this field is assigned with a value, along with a ticket number?

Reply:
Fair enough: [74d92d2](https://github.com/cardano-foundation/cardano-wallet/pull/4538/commits/74d92d26962a66bece2384761c5939f61c0c3a30)

Reply:
Great, thanks!